### ISSUE TITLE
[Python] Implement new Concept Exercise: comparison

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#2039]
This issue describes how to implement the concept exercise, for the Python track, which should explain how comparisons works in python.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should convey a basic understanding of the comparison's operations in python. A comparison operator is a construct (or operator) that tests or defines some kind of relation between two entities: integers, objects...

## Learning objectives

- use comparisons operators to check for numerical equality and inequalities `==`, `>`, `<`, `!=`
- understand (and use the concept) that the `==` operator calls the dunder method `__eq__()`, use this comparison methods to compare objects
- overload the default implementation of the `__eq__` dunder method

## Out of scope

- rich comparison with `__lt__`, `__le__`, `__ne__`, `__ge__`, `__gt__`

## Concepts

- Comparison operators`==`, `>`, `<`, `!=`
- Integer comparison
- Rich comparison methods `__eq__`
- Equality operator
- Equivalence
- Inequality

## Prerequisites

- Order of evaluation
- Dunder methods
- Overload

## Resources to refer to

- [Python basic operators on tutorialspoint](https://www.tutorialspoint.com/python/python_basic_operators.htm)
- [Python comparison operators on data-flair](https://data-flair.training/blogs/python-comparison-operators/)
- [Python comparison operators on data-flair](https://data-flair.training/blogs/python-comparison-operators/)
- [Python rich comparison official doc](https://docs.python.org/3/library/operator.html)

### Hints

- TBD?

### After

Nothing

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named bool_basic_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
-

**Comment by DavidGerva** [comment]
@BethanyG would like to have a review to this issue. Not sure about:
- the rich comparison inside this exercise
- the english form and use of the word "comparison"

**Comment by BethanyG** [comment]
@DavidGerva  - apologies for taking so long to give you feedback.  Thank you so much for filing this issue and for raising questions about it.

 I really struggled to zero in on what felt like the "right" collection of functionality here.  I am still not sure I got it correct, and we may need to discuss this further with others.  As for language -- I changed the title to `comparisons` on the logic that Python has many methods and implementations of comparison, and we are talking specifically about those and not the general concept. 

I think that `numbers` and `floating-point-numbers` - or some variants thereof might include comparisons, but that it's fine to have overlap.  I think making _numeric comparison_ its own exercise is probably overkill.  Still not sure that `None` and `NotImplemented` belong -- we may want to remove those.

I added more comparisons for various types, since the Python docs point out that many built-in types have their own comparison rules that might not be intuitive at first glance.  I also added an item about comparisons yielding `boolean` values since we pulled comparison operators out of the `booleans` exercise.  I essentially followed [this comparisons entry](https://docs.python.org/3/reference/expressions.html#comparisons)  in the Python language reference, but with a lot less detail.  

 I moved the `__eq__()` method and any customization to "out of scope" -- I think language and behavior customizations should be their own exercise -- or at the very least, we should have a "rich comparisons" exercise where we might cover customization. 

Having said all that - I am very open to pushback.  I could see equal arguments for including comparisons with each built-in type or for pulling it out into an exercise as we have here.

@cmccandless  @ErikSchierboom  - your thoughts?  are we on the right track??

**Comment by ErikSchierboom** [comment]
> I think that numbers and floating-point-numbers - or some variants thereof might include comparisons, but that it's fine to have overlap. I think making numeric comparison its own exercise is probably overkill. Still not sure that None and NotImplemented belong -- we may want to remove those.

Some overlap is fine here I feel.

> I moved the __eq__() method and any customization to "out of scope" -- I think language and behavior customizations should be their own exercise -- or at the very least, we should have a "rich comparisons" exercise where we might cover customization.

Without having any Python knowledge, it _sounds_ like a sensible thing to do.

**Comment by cmccandless** [comment]
Sounds good to me!

**Comment by BethanyG** [comment]
Oh for &^!#% sake.  None of the edits I made to this issue were actually saved.  **I am so annoyed right now**.  Apologies everyone for the spam with no purpose.  Will re-edit.

**Comment by BethanyG** [comment]
Re-edited with details described above.  Chime in if this looks strange..

**Comment by DavidGerva** [comment]
Really nice. I'm glad it was useful to raise the doubts.

**Comment by Limm-jk** [comment]
Hello!
Is this issue in progress?
If not, can I take up the issue?

**Comment by BethanyG** [comment]
Hi @Limm-jk !  We'd be delighted if you wanted to take up this issue 🌟   **Thank you.**  🌈 

I don't yet have a concept description (_apologies for that_), but can work on one today/this weekend & post it here when I'm done.  LMK here or in Slack if you have any questions or problems.  

Looking at the list of items to cover, it does look long - so we can have a discussion if it looks like the exercise is going to get too complex.  Since these are comparisons (and return booleans), it might be fun to do an extension on the story for bools that we have here:  [Ghost Gobble Arcade Game](https://github.com/exercism/v3/tree/master/languages/python/exercises/concept/ghost-gobble-arcade-game) -- but you can **always** come up with your own story 😄 , or you can also use one from this list: [exercise stories](https://github.com/exercism/v3/tree/master/reference/stories).  

_Thanks again for volunteering!_

**Comment by Limm-jk** [comment]
Thank you! 
I'll get started as soon as you tell me.
Where can I get information about exercism's slacks?
I would appreciate it if you could tell me how to work in Slack.

**Comment by BethanyG** [comment]
Hi @Limm-jk!  For Slack, I just need an email and I can send you an invite directly.  Is the one on your GitHub profile OK to use?

**Comment by BethanyG** [comment]
@Limm-jk  - also - it is ok to start writing now if you are eager to get started.  I may not have the concept description written until tomorrow my time (I'm in Pacific time, so I am 17 hours behind you.  😆 ).

**Comment by Limm-jk** [comment]
> Hi @Limm-jk! For Slack, I just need an email and I can send you an invite directly. Is the one on your GitHub profile OK to use? I will send an invite there, and we can change it later if you need to.
> 
> EDIT: Since invites need admin approval, and the admins are in GMT and CET, there will be a delay for Slack.

Good!
